### PR TITLE
[structure] Add pure Neumann test case for Euler-Bernoulli and Simo-Reissner beams.

### PR DESCRIPTION
**Pull request by maxfirmbach**
## Description and Context

This PR adds two test cases for Euler-Bernoulli beams with only Neumann boundary conditions. The system is singular due to the unconstrained rigid body modes, which are removed by using a Krylov projection of the nullspace.

The resulting projection operators are dense, resulting in a dense system matrix. Therefore this approach is only feasible for small system sizes.

### CODE REVIEW

File: src/global_legacy_module/4C_global_legacy_module_validconditions.cpp [hunk 1]
```diff
@@ -785,6 +785,11 @@ std::vector<Core::Conditions::ConditionDefinition> Global::valid_conditions()
   // 3-D solid, 3 for 2-D solid), where ONOFF triggers first the
   // translational followed by the rotational modes, each in/around x to z
 
+  Core::Conditions::ConditionDefinition linerigidbodymode(
+      "DESIGN LINE MODE FOR KRYLOV SPACE PROJECTION", "KrylovSpaceProjection",
+      "Surface mode for Krylov space projection", Core::Conditions::LineModeKrylovProjection, true,
```
Comment (mayrmt) [inline src/global_legacy_module/4C_global_legacy_module_validconditions.cpp]:
```suggestion
      "Line mode for Krylov space projection", Core::Conditions::LineModeKrylovProjection, true,
```

File: src/structure_new/src/linear_solver/4C_structure_new_solver_factory.cpp [hunk 2]
```diff
@@ -163,8 +164,16 @@ std::shared_ptr<Core::LinAlg::Solver> Solid::SOLVER::Factory::build_structure_li
 
       Core::LinAlg::Map nullspace_map(*actdis.dof_row_map());
 
+      int numdof;
+      int dimns;
+      {
+        // just grab the block information on the first element that appears
+        Core::Elements::Element* dwele = actdis.l_row_element(0);
+        dwele->element_type().nodal_block_information(dwele, numdof, dimns);
+      }
```
Comment (mayrmt) [inline src/structure_new/src/linear_solver/4C_structure_new_solver_factory.cpp]: Does this mean, that this only works if all elements in the discretization are of the same type?

Comment (maxfirmbach) [reply src/structure_new/src/linear_solver/4C_structure_new_solver_factory.cpp]: In the current state yes, but this assumption also holds in other parts of the code.

File: src/global_legacy_module/4C_global_legacy_module_validconditions.cpp (hunk 1 above)
Comment (maxfirmbach) [reply src/global_legacy_module/4C_global_legacy_module_validconditions.cpp]: Did rewrite the condition description.

File: src/structure_new/src/linear_solver/4C_structure_new_solver_factory.cpp [hunk 3]
```diff
@@ -137,6 +137,7 @@ std::shared_ptr<Core::LinAlg::Solver> Solid::SOLVER::Factory::build_structure_li
       const int nummodes = kspcond->parameters().get<int>("NUMMODES");
 
       // get rigid body mode flags for a 3-D solid: [transx transy transz rotx roty rotz]
+      // Euler-Bernoulli beams have a similar ordering: [transx transy transz rot1 rot2]
```
Comment (dharinib98) [inline src/structure_new/src/linear_solver/4C_structure_new_solver_factory.cpp]: What about rot3 ?

Comment (mayrmt) [reply src/structure_new/src/linear_solver/4C_structure_new_solver_factory.cpp]: Two rotations are fine for Euller-Bernoulli. Details can be found in the bachelor thesis by Max Bahnmüller.

**Comment by maxfirmbach** [comment]
@isteinbrecher Maybe you can have a look at the load linearization flag and if the proposed changes make sense.

**Comment by maxfirmbach** [comment]
I put this in draft-mode for now. I will also add the related tests for Simo-Reissner beams with this PR soon.

**Comment by isteinbrecher** [comment]
> @isteinbrecher Maybe you can have a look at the load linearization flag and if the proposed changes make sense.

I am not sure if they make sense. Why do you have to change this, this isn't completely clear to me from the changes.

File: src/structure_new/src/model_evaluator/4C_structure_new_model_evaluator_structure.cpp [hunk 4]
```diff
@@ -381,12 +381,15 @@ bool Solid::ModelEvaluator::Structure::apply_force_stiff_external()
   if (eval_data().get_damping_type() == Inpar::Solid::damp_material)
     discret().set_state(0, "velocity", *global_state().get_vel_n());
 
+  // TODO: is it okay to move the set_state() outside of the if-else?
+  discret().set_state(0, "displacement new", *global_state().get_dis_np());
```
Comment (mayrmt) [inline src/structure_new/src/model_evaluator/4C_structure_new_model_evaluator_structure.cpp]: @maxfirmbach I am ok with moving it here.

File: src/structure_new/src/model_evaluator/4C_structure_new_model_evaluator_structure.cpp [hunk 5]
```diff
@@ -381,12 +381,15 @@ bool Solid::ModelEvaluator::Structure::apply_force_stiff_external()
...
+
   // get load vector
   if (!tim_int().get_data_sdyn().get_load_lin())
     evaluate_neumann(*global_state().get_fext_np(), nullptr);
   else
   {
-    discret().set_state(0, "displacement new", *global_state().get_dis_np());
+    // discret().set_state(0, "displacement new", *global_state().get_dis_np());
```
Comment (mayrmt) [inline src/structure_new/src/model_evaluator/4C_structure_new_model_evaluator_structure.cpp]:
```suggestion
```

File: src/structure_new/src/model_evaluator/4C_structure_new_model_evaluator_structure.cpp (hunk 4 above)
Comment (dharinib98) [reply src/structure_new/src/model_evaluator/4C_structure_new_model_evaluator_structure.cpp]: If we move `set_state()` outside the if-else, we would not need the updated error message from #1219

**Comment by maxfirmbach** [comment]
As discussed offline with @isteinbrecher, I set the load linearization for all `beam3eb` test cases to true. This also means that the changes to the setting of the displacement state is not necessary anymore.

**Comment by isteinbrecher** [comment]
From what I can see, the changes look good to me.

**Comment by maxfirmbach** [comment]
All checks have passed, so looking good.

**Review by mayrmt — APPROVED** [review]

File: tests/input_files/beam3r_herm2line3_static_one_element_pure_neumann_projection.4C.yaml [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+  - "NODE 3 COORD 2.0 0.0 0.0"
+STRUCTURE ELEMENTS:
+  - "1 BEAM3R LINE3 1 3 2 HERMITE_CENTERLINE true MAT 1 TRIADS 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0"
+RESULT DESCRIPTION:
+  - STRUCTURE:
+      DIS: "structure"
+      NODE: 1
+      QUANTITY: "dispx"
+      VALUE: 0.0
+      TOLERANCE: 1e-5
```
Comment (isteinbrecher) [inline tests/input_files/beam3r_herm2line3_static_one_element_pure_neumann_projection.4C.yaml:172]: Why the higher tolerance for x?

Comment (maxfirmbach) [reply tests/input_files/beam3r_herm2line3_static_one_element_pure_neumann_projection.4C.yaml:172]: Seems like the displacement there is not exactly 0 yet.

Comment (isteinbrecher) [reply tests/input_files/beam3r_herm2line3_static_one_element_pure_neumann_projection.4C.yaml:172]: If you say that is fine, let's go ahead with this, it is just something I noticed.

Comment (maxfirmbach) [reply tests/input_files/beam3r_herm2line3_static_one_element_pure_neumann_projection.4C.yaml:172]: @isteinbrecher I would say let's go for the merge for now.

**Comment by maxfirmbach** [comment]
I have to rebase to resolve the merge conflicts ...
